src/reconcile/rec_row.c

```diff
@@ -327,6 +329,13 @@ __wti_rec_row_int(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
 
     /* For each entry in the in-memory page... */
     WT_INTL_FOREACH_BEGIN (session, page, ref) {
+        /*
+         * Set the ref_changes state to zero if there were no concurrent changes while reconciling the
+         * internal page.
+         */
+        prev_ref_changes = ref->ref_changes;
+        __wt_atomic_casv16(&ref->ref_changes, prev_ref_changes, 0);
```

Comment:
We should read the ref at the start of the loop and cas at the end of the loop not immediately.